Converted to Companion v3

Engma90 opened the pull request:
- [x] Converted to Companion v3
- [x] Applied code formatting as mentioned in this [instructions page](https://github.com/bitfocus/companion-module-base/wiki/Code-formatting)

Engma90 committed:
Converted to Companion v3

- Still need to Find an alternative to this.system in v3
- Testing required

Engma90 committed:
Removed old variable sender

the this.system was used and it is no longer supported in v3

Engma90 committed:
Passed variable to CueTimer using Trigger & Action

Engma90 committed:
Added & applied code linting

Engma90 committed:
Temporarily removed send variables feature

It will be added in later releases

dnmeid commented on package.json:
```diff
@@ -1,38 +1,27 @@
-	"version": "1.1.4",
-	"api_version": "1.0.0",
+	"version": "2.0.0",
 	"support": {
```
this isn't needed or used. You can safely remove it.

Engma90 replied:
Done, 
BTW, I just didn't remove it because it wasn't mentioned in`package.json cleanup` section [here](https://github.com/bitfocus/companion-module-base/wiki/Upgrading-a-module-built-for-Companion-2.x#3-packagejson-cleanup)

krocheck replied:
@Engma90 there was some minor confusion.  The `version` field is required in package.json.  The [manifest.json](https://github.com/bitfocus/companion-module-base/wiki/manifest.json) page explains this.  Can you please add it back in.

Engma90 replied:
No worries, it is added now

dnmeid commented on package.json:
```diff
@@ -1,38 +1,27 @@
 		"url": "git+https://github.com/bitfocus/companion-module-presentationtools-cuetimer.git"
+	},
+	"dependencies": {
+		"@companion-module/base": "~1.1.0"
```
Suggest update to latest version

Engma90 replied:
Done

dnmeid commented on package.json:
```diff
@@ -1,38 +1,27 @@
+		"@companion-module/base": "~1.1.0"
+	},
+	"devDependencies": {
+		"@companion-module/tools": "^0.5.2"
```
Suggest update to latest version

Engma90 replied:
Done

dnmeid commented on yarn.lock:
```diff
@@ -0,0 +1,2107 @@
+  integrity sha512-/DyMEOrDgLKKIG0fmvtz+4dUX/3Ghozwgm6iPp8KRhvn+eQf9+Q7GWxVNMk3+uCPWfdXYC4ExGBckIXdFEfH1w==
+
+webpack@^5.75.0:
+  version "5.75.0"
```
Should be 5.76.1 due to vulnerability.

Engma90 replied:
I am not sure how I can update this, it is a dependency of `@companion-module/tools`

krocheck replied:
Dependabot will handle this after merge.

dnmeid reviewed (changes requested):
Gave some comments in code. No showstoppers, just small things.

Engma90 committed:
Fixed issues

- Used correct linting config
- upgraded @companion-module/base & @companion-module/tools
- Removed version from package.json

Engma90 committed:
Added version

krocheck commented on index.js:
```diff
@@ -1,634 +1,465 @@
-	}
-	self.timers = {}
-}
+	updateConfig(config) {
```
This should be `async configUpdated`

Engma90 replied:
Done

krocheck commented on index.js:
```diff
@@ -1,634 +1,465 @@
+					let jsonData = JSON.parse(message)
+					let hours = (jsonData.h < 0 ? '+' : '') + jsonData.h.replace('-', '')
+					hours = hours == '' ? '' : hours.lpad('0', 2)
+					self.setVariableValues({ hours: hours })
```
I feel like this `setVariableValues` for hours, as well as the ones for minutes and seconds, could be rolled into the later one that sets speed, name, and endTime

Engma90 replied:
I think you are right

Engma90 replied:
Done

Engma90 committed:
Made updateConfig async

krocheck commented on index.js:
```diff
@@ -1,634 +1,465 @@
-					label: 'ID',
-					id: 'Key',
-					default: '1',
+			timerName: {
```
I would eventually like to see the text in this advanced feedback (and timerDuration) also setup as a variables.

Engma90 replied:
If I got you right, I think you mean giving the user the ability to show these values by using variables in addition to feedback. but the problem is these values are linked to a specific timer from the timers list with its id which is dynamically entered by the user (please check the attached image) 
In feedback, this can be fixed by passing the id in options, but I am not sure how this can be done with variables

BTW, we already have variables for active & next timers but for the rest we used feedback

![image](https://user-images.githubusercontent.com/12226980/226204835-34abc3e7-0b29-4e1f-b873-b80392b77a6c.png)

krocheck replied:
Without putting a ton of effort/thought in, I think my understanding of your dataset would yield this type of implementation: https://github.com/bitfocus/companion-module-presentationtools-cuetimer/commit/4106a9ff094193a8450517f62690a1037d983c1f

krocheck replied:
It would be nice in that scenario to check the keys in `jsonData.timers` vs `self.timers` before storing to figure out if `self.variables()` is required to run.

Engma90 replied:
OK, it is easier than I thought, will merge your branch into this branch and add that keys check too, thanks!

Engma90 replied:
Please check 4639f650e66aa28c25970f3c77a9f810f1819bea and let me know if this is ok or I missed anything

krocheck replied:
Yeah, did it test OK with the software?

Engma90 replied:
We just finished testing, sorry for the delay.  
Variables are working fine, we just removed the bg color it will only show the hex code.

So now if you think it is fine, we are ready to merge and add to the core. Thanks!

Engma90 committed:
updated hours, minutes, seconds variables in one statement

Engma90 committed:
Corrected configUpdated function name

krocheck committed:
Timer variables first pass

Engma90 committed:
Added a check before calling self.variables()

Engma90 committed:
Removed background from dynamic timer variables
